be/src/exec/vectorized/json_scanner.cpp

```diff
@@ -715,80 +735,209 @@ void JsonReader::_construct_column(simdjson::ondemand::value& value, Column* col
     }
 
     case simdjson::ondemand::json_type::number: {
-        simdjson::ondemand::number_type tp;
+        _construct_number_column(value, column, type_desc);
+        break;
+    }
+
+    case simdjson::ondemand::json_type::string: {
+        _construct_string_column(value, column, type_desc);
+        break;
+    }
+
+    case simdjson::ondemand::json_type::object:
+    case simdjson::ondemand::json_type::array: {
+        std::unique_ptr<char[]> buf;
+        size_t buflen{};
+        JsonFunctions::minify_json_to_string(value, buf, buflen);
+
+        column->append_strings(std::vector<Slice>{Slice{buf.get(), buflen}});
+        break;
+    }
+
+    default: {
+        column->append_nulls(1);
+        break;
+    }
+    }
+}
 
-        auto err = value.get_number_type().get(tp);
+void JsonReader::_construct_number_column(simdjson::ondemand::value& value, Column* column,
+                                          const TypeDescriptor& type_desc) {
+    simdjson::ondemand::number_type tp;
+
+    auto err = value.get_number_type().get(tp);
+    if (UNLIKELY(err)) {
+        column->append_nulls(1);
+        return;
+    }
+
+    if (tp == simdjson::ondemand::number_type::signed_integer) {
+        int64_t i64 = 0;
+        err = value.get_int64().get(i64);
         if (UNLIKELY(err)) {
             column->append_nulls(1);
-            break;
+            return;
         }
 
-        switch (tp) {
-        case simdjson::ondemand::number_type::signed_integer: {
-            int64_t i64;
-            err = value.get_int64().get(i64);
-            if (UNLIKELY(err)) {
-                column->append_nulls(1);
-                break;
-            }
+        switch (type_desc.type) {
+        case TYPE_BIGINT: {
+            column->append_numbers(&i64, sizeof(i64));
```

Comment:
Return the return value or using `down_cast` to case the column to FixedLengthColumn<int64_t>

Reply:
Updated.